Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -15,57 +15,78 @@
+        public ProjectArea ProjectArea { get; private set; }
 
         public RestConnector mOctaneRestConnector;
         public EntityService entityService;
         protected LwssoAuthenticationStrategy lwssoAuthenticationStrategy;
         protected WorkspaceContext workspaceContext;
         protected SharedSpaceContext sharedSpaceContext;
+        protected OctaneRepository octaneRepository;
 
         public override bool ConnectALMProject()
         {
-            throw new NotImplementedException();
+            return this.ConnectALMServer();
         }
 
         public override bool ConnectALMServer()
         {
             try
             {
+                if (octaneRepository == null)
+                    octaneRepository = new OctaneRepository();
                 Reporter.ToLog(eLogLevel.DEBUG, "Connecting to Octane server");
-                mOctaneRestConnector = new RestConnector();
-                entityService = new EntityService(mOctaneRestConnector);
-                if (!mOctaneRestConnector.IsConnected())
-                {
-                    string ignoreServerCertificateValidation = "false"; //ConfigurationManager.AppSettings["ignoreServerCertificateValidation"];
-                    if (ignoreServerCertificateValidation != null && ignoreServerCertificateValidation.ToLower().Equals("true"))
+                return Task.Run(() =>
                     {
-                        NetworkSettings.IgnoreServerCertificateValidation();
-                    }
-                    NetworkSettings.EnableAllSecurityProtocols();                  
-
-                    // If webAppUrl is empty we do not try to connect.
-                    if (string.IsNullOrWhiteSpace(ALMCore.DefaultAlmConfig.ALMServerURL)) return false;
-
-                    IConnectionInfo connectionInfo;
-     
-                    connectionInfo = new UserPassConnectionInfo(ALMCore.DefaultAlmConfig.ALMUserName, ALMCore.DefaultAlmConfig.ALMPassword);
-                
-                    lwssoAuthenticationStrategy = new LwssoAuthenticationStrategy(connectionInfo);
-                    var result =  Task.Run(() =>
-                    {
-                        return  mOctaneRestConnector.Connect(ALMCore.DefaultAlmConfig.ALMServerURL, lwssoAuthenticationStrategy);
-                     
-                    });
-                    return result.Result;       
-
-                }
+                        return octaneRepository.IsLoginValid(
+                            new LoginDTO()
+                            {
+                                User = ALMCore.DefaultAlmConfig.ALMUserName,
+                                Password = ALMCore.DefaultAlmConfig.ALMPassword,
+                                Server = ALMCore.DefaultAlmConfig.ALMServerURL
+                            });
+                    }).Result;
+                //mOctaneRestConnector = new RestConnector();
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6235433)